src/types.rs

```diff
@@ -450,6 +459,52 @@
     }
 }
 
+pub struct SubjectBuilder {
+    tokens: Vec<String>
+}
+
+impl SubjectBuilder {
+    pub fn new() -> Self {
+        SubjectBuilder {
+            tokens: Vec::new()
+        }
+    }
+
+    pub fn add(mut self, subject: String) -> Self {
```

Comment:
Could you make this function signiture `pub fn add(mut self, subject: impl Into<String>) -> Self`? That way you could use `&str` or `String`.